src/Interfacer.jl

```diff
@@ -101,6 +101,14 @@ abstract type SeaIceModelSimulation <: SurfaceModelSimulation end
 abstract type LandModelSimulation <: SurfaceModelSimulation end
 abstract type OceanModelSimulation <: SurfaceModelSimulation end
 
+"""
+    getfield(sim::ComponentModelSimulation, val::Val)
+
+A getter function that should not allocate. Here we implement a default that
+will raise an error if `get_field` isn't defined for all required fields of
+a component model.
+"""
+get_field(sim::ComponentModelSimulation, val::Val) = error("undefined field $val for " * name(sim))
```

Comment:
good idea! Since `AtmosModelSimulation` and `SurfaceModelSimulation` have different requirements, I split this up into two separate methods dispatching on those types.

Reply:
Perfect, thanks!